src/openloco/windows/construction/ConstructionTab.cpp

```diff
@@ -0,0 +1,1280 @@
+
+                _byte_508F09 = _byte_508F09 & ~(1 << 0);
+
+                if (_dword_1135F42 == 0x80000000)
+                {
+                    if (gGameCommandErrorText != string_ids::error_can_only_build_above_ground)
+                    {
+                        ebx--;
+                        if (ebx != 0)
+                        {
+                            trackHeight -= 16;
+                            if (trackHeight >= 0)
+                            {
+                                if (ebx < 0)
+                                {
+                                    continue;
+                                }
+                                else
+                                {
+                                    trackHeight += 32;
+                                    continue;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    _byte_113607E = 1;
+                    WindowManager::close(WindowType::error, 0);
+                    return;
+                }
+
+                on_mouse_up(window, widx::rotate_90);
+
+                audio::play_sound(audio::sound_id::error, int32_t(input::getMouseLocation().x));
+
+                return;
+            }
+        }
+    }
+
+    // 0x0049D4F5
+    static ui::cursor_id cursor(window* self, int16_t widgetIndex, int16_t xPos, int16_t yPos, ui::cursor_id fallback)
+    {
+        if (widgetIndex == widx::bridge || widgetIndex == widx::bridge_dropdown)
+            _tooltipTimeout = 2000;
+        return fallback;
+    }
+
+    // 0x0049CE79
+    static void prepare_draw(window* self)
+    {
+        common::prepare_draw(self);
+        auto args = FormatArguments();
+        if (_trackType & (1 << 7))
+        {
+            auto roadObj = objectmgr::get<road_object>(_trackType & ~(1 << 7));
+            args.push(roadObj->name);
+        }
+        else
+        {
+            auto trackObj = objectmgr::get<track_object>(_trackType);
+            args.push(trackObj->name);
+        }
+        if (_lastSelectedBridge != 0xFF)
+        {
+            auto bridgeObj = objectmgr::get<bridge_object>(_lastSelectedBridge);
+            if (bridgeObj != nullptr)
+            {
+                args.push(bridgeObj->name);
+                if (bridgeObj->max_speed == 0xFFFF)
+                {
+                    args.push(string_ids::unlimited_speed);
+                    args.push<uint16_t>(0);
+                }
+                else
+                {
+                    args.push(string_ids::velocity);
+                    args.push(bridgeObj->max_speed);
+                }
+                args.push<uint16_t>(bridgeObj->max_height);
+            }
+        }
+        common::repositionTabs(self);
+    }
+
+    // 0x004A0AE5
+    void drawTrack(uint16_t x, uint16_t y, uint32_t edi, uint8_t bh, uint32_t edx)
```

Comment:
edi is clearly two separate parameters. Do the combination in this function so the callers can be clean

Reply:
And edx is three parameters. Just split them all up so the callers are nice and clean.